scripts/core/editor3/components/annotations/AnnotationInput.tsx

```diff
@@ -75,12 +72,24 @@ class AnnotationInputBody extends React.Component<IProps, IState> {
             body: body != null ? body : convertToRaw(ContentState.createFromText('')),
             type: type,
             isEmpty: isEmpty,
+            loaded: false,
         };
 
         this.onSubmit = this.onSubmit.bind(this);
         this.onChange = this.onChange.bind(this);
         this.onSelect = this.onSelect.bind(this);
         this.deleteAnnotation = this.deleteAnnotation.bind(this);
+        this.getSelectionText = this.getSelectionText.bind(this);
+
+        this.annotationInputTabsFromExtensions = flatMap(
```

Comment:
imo it would be good to have some api to work with extensions, so you don't have to import that generated file everywhere and do the map on it, something like:
```
import extensions from 'core/extensions';

this.annotationInputTabsFromExtensions = extensions.getFlatMap('contributions.editor3.annotationInputTabs');
```
so it will be simpler to write tests for code allowing extensions

Reply:
I agree, but I don't think it's possible to do it in a straight-forward way and also preserve static-typing. It generates a few lines of boring boilerplate, but there's no complexity there, so I'm not worried about it.